app/navigation/ExpensesNavigator.scala

```diff
@@ -49,195 +42,151 @@ import javax.inject.{Inject, Singleton}
-              userAnswers.get(AdvertisingOrMarketingPage, Some(businessId)) match {
-                case Some(AdvertisingOrMarketing.YesDisallowable) if accountingType.contains(Accrual) =>
-                  advertisingOrMarketing.routes.AdvertisingDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
-                case Some(_) => advertisingOrMarketing.routes.AdvertisingCYAController.onPageLoad(taxYear, businessId)
-                case None    => standard.routes.JourneyRecoveryController.onPageLoad()
-              }
+            userAnswers.get(AdvertisingOrMarketingPage, Some(businessId)) match {
+              case Some(AdvertisingOrMarketing.YesDisallowable) =>
+                advertisingOrMarketing.routes.AdvertisingDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
+              case Some(AdvertisingOrMarketing.YesAllowable) => advertisingOrMarketing.routes.AdvertisingCYAController.onPageLoad(taxYear, businessId)
+              case _                                         => standard.routes.JourneyRecoveryController.onPageLoad()
+            }
 
     case AdvertisingOrMarketingDisallowableAmountPage =>
-      _ => taxYear => businessId => _ => advertisingOrMarketing.routes.AdvertisingCYAController.onPageLoad(taxYear, businessId)
+      _ => taxYear => businessId => advertisingOrMarketing.routes.AdvertisingCYAController.onPageLoad(taxYear, businessId)
 
     case EntertainmentAmountPage =>
-      _ => taxYear => businessId => _ => entertainment.routes.EntertainmentCYAController.onPageLoad(taxYear, businessId)
-
-    case EntertainmentCYAPage =>
-      _ =>
-        taxYear =>
-          businessId =>
-            _ => journeys.routes.SectionCompletedStateController.onPageLoad(taxYear, businessId, ExpensesEntertainment.toString, NormalMode)
+      _ => taxYear => businessId => entertainment.routes.EntertainmentCYAController.onPageLoad(taxYear, businessId)
 
     case ConstructionIndustryAmountPage =>
       userAnswers =>
         taxYear =>
           businessId =>
-            accountingType =>
-              userAnswers.get(DisallowableSubcontractorCostsPage, Some(businessId)) match {
-                case Some(DisallowableSubcontractorCosts.Yes) if accountingType.contains(Accrual) =>
-                  construction.routes.ConstructionIndustryDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
-                case Some(_) =>
-                  construction.routes.ConstructionIndustryCYAController.onPageLoad(taxYear, businessId)
-                case None => standard.routes.JourneyRecoveryController.onPageLoad()
-              }
+            userAnswers.get(DisallowableSubcontractorCostsPage, Some(businessId)) match {
+              case Some(DisallowableSubcontractorCosts.Yes) =>
+                construction.routes.ConstructionIndustryDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
+              case Some(_) =>
```

Comment:
Could we maybe use `DisallowableSubcontractorCosts.No` here just to be more specific in our matching?